nemo/collections/asr/parts/utils/diarization_utils.py

```diff
@@ -986,41 +1013,45 @@ def print_errors(self, DER_result_dict, WDER_dict):
                 \nSpk. counting acc.: {DER_result_dict['total']['spk_counting_acc']:.4f}"
             )
 
-    def print_time(self, speaker, start_point, end_point, params, previous_string=None, replace_time=False):
+    def print_sentences(self, sentences, params):
         """
         Print a transcript with speaker labels and timestamps.
         """
-        if not previous_string:
-            string_out = ''
-        else:
-            string_out = previous_string
-        if params['colored_text']:
-            color = self.color_palette.get(speaker, '\033[0;37m')
-        else:
-            color = ''
+        # init output
+        string_out = ''
+
+        for sentence in sentences:
+            # extract info
+            speaker = sentence['speaker']
+            start_point = sentence['start_point']
+            end_point = sentence['end_point']
+            text = sentence['text']
+
+            # color?
+            if params['colored_text']:
+                color = self.color_palette.get(speaker, '\033[0;37m')
+            else:
+                color = ''
 
-        datetime_offset = 16 * 3600
-        if float(start_point) > 3600:
-            time_str = "%H:%M:%S.%f"
-        else:
-            time_str = "%M:%S.%f"
-        start_point, end_point = max(float(start_point), 0), max(float(end_point), 0)
-        start_point_str = datetime.fromtimestamp(start_point - datetime_offset).strftime(time_str)[:-4]
-        end_point_str = datetime.fromtimestamp(end_point - datetime_offset).strftime(time_str)[:-4]
-
-        if replace_time:
-            old_start_point_str = string_out.split('\n')[-1].split(' - ')[0].split('[')[-1]
-            word_sequence = string_out.split('\n')[-1].split(' - ')[-1].split(':')[-1].strip() + ' '
-            string_out = '\n'.join(string_out.split('\n')[:-1])
-            time_str = "[{} - {}] ".format(old_start_point_str, end_point_str)
-        else:
-            time_str = "[{} - {}] ".format(start_point_str, end_point_str)
-            word_sequence = ''
+            # cast timestamp to the correct format
+            datetime_offset = 16 * 3600
+            if float(start_point) > 3600:
+                time_str = '%H:%M:%S.%f'
+            else:
+                time_str = '%M:%S.%f'
+            start_point, end_point = max(float(start_point), 0), max(float(end_point), 0)
+            start_point_str = datetime.fromtimestamp(start_point - datetime_offset).strftime(time_str)[:-4]
+            end_point_str = datetime.fromtimestamp(end_point - datetime_offset).strftime(time_str)[:-4]
+
+            # print time?
```

Comment:
Hi tango4j!

Time replacement is now handled inside the `make_json_output` function (line 529). If the speaker is the same as the previous one we do not have a new sentence but the end time gets tweaked. The change is then also reflected in the printed out time (`string_out` variable that gets printed out to the .txt file) automatically. There is no need to have the `replace_time` argument there.

I might be wrong since I do not know how exactly the streaming pipeline will work. I can revive the  `replace_time` argument, but in the current codebase it would not make any sense. I am writing this so we are on the same page. Please advise about how to proceed here.

Thanks, Jure.

Reply:
Now I understand your point. Thanks.
I made a small request in a separate review comment.